### PR TITLE
feat(blockbuilder): grpc transport

### PR DESCRIPTION
**Pull request by owen-d**
gRPC transport implementation as defined in https://github.com/grafana/loki/blob/c1ca06793d561a708fa9ff38f7e2fc012c90b064/pkg/blockbuilder/architecture.md

### CODE REVIEW

File: pkg/blockbuilder/types/grpc_transport.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health/grpc_health_v1"
+
+	"github.com/grafana/loki/v3/pkg/blockbuilder/types/proto"
+	"github.com/grafana/loki/v3/pkg/util/constants"
+)
+
+var _ Transport = &GRPCTransport{}
+
+type GRPCTransportConfig struct {
```
Comment (ashwanthgoli) [inline pkg/blockbuilder/types/grpc_transport.go:22]: nit: should we embed this in builder config? ignore me if it is planned for a follow-up

File: pkg/blockbuilder/types/grpc_transport.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	Address string `yaml:"address,omitempty"`
+
+	// GRPCClientConfig configures the gRPC connection between the Bloom Gateway client and the server.
```
Comment (ashwanthgoli) [inline pkg/blockbuilder/types/grpc_transport.go:25]:
```suggestion
	// GRPCClientConfig configures the gRPC connection between the block builder client and the schedulers.
```

**Review by ashwanthgoli — APPROVED** [review]
lgtm!

File: pkg/blockbuilder/types/grpc_transport.go (hunk 1 above)
Comment (owen-d) [reply pkg/blockbuilder/types/grpc_transport.go:22]: Yeah, I planned to leave this for when we're ready to wire it together.

**Pull request merged**